src/api/VoteMonitor.Api.PollingStation/Controllers/PollingStationV2Controller.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System.Collections.Generic;
+using System.Threading.Tasks;
+using AutoMapper;
+using MediatR;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using VoteMonitor.Api.PollingStation.Models;
+using VoteMonitor.Api.PollingStation.Queries;
+
+namespace VoteMonitor.Api.PollingStation.Controllers
+{
+    /// <summary>
+    /// Controller responsible for interacting with the polling stations - PollingStationInfo 
+    /// </summary>
+    [Route("api/v2/polling-station")]
+    public class PollingStationV2Controller : Controller
+    {
+        private readonly IMediator _mediator;
+        private readonly IMapper _mapper;
+
+        public PollingStationV2Controller(IMediator mediator, IMapper mapper)
+        {
+            _mediator = mediator;
+            _mapper = mapper;
+        }
+
+        [HttpGet]
+        [Produces(typeof(IEnumerable<GetPollingStation>))]
+        public async Task<IActionResult> GetAllPollingStations([FromQuery]PollingStationsFilter pollingStationsFilter)
+        {
+            //todo fix paging model defaults
+            var request = _mapper.Map<GetPollingStations>(pollingStationsFilter);
+
+            var result = await _mediator.Send(request);
+
+            return Ok(result);
+        }
+
+        [HttpPut("{id}")]
+        [Authorize("NgoAdmin")]
```

Comment:
Please use `[Authorize("Organizer")]` here to restrict this to master admins :smile:

Reply:
Done